internal/resources/syntheticmonitoring/resource_check_alert.go

```diff
@@ -0,0 +1,226 @@
+		common.CategorySyntheticMonitoring,
+		"grafana_synthetic_monitoring_check_alerts",
+		resourceCheckAlertID,
+		&schema.Resource{
+			Description: `
+Manages alerts for a check in Grafana Synthetic Monitoring.
+
+* [Official documentation](https://grafana.com/docs/grafana-cloud/synthetic-monitoring/configure-alerts/)
+* [API documentation](https://github.com/grafana/synthetic-monitoring-api-go-client/blob/main/docs/API.md#alerts)
+`,
+
+			CreateContext: withClient[schema.CreateContextFunc](resourceCheckAlertCreate),
+			ReadContext:   withClient[schema.ReadContextFunc](resourceCheckAlertRead),
+			UpdateContext: withClient[schema.UpdateContextFunc](resourceCheckAlertUpdate),
+			DeleteContext: withClient[schema.DeleteContextFunc](resourceCheckAlertDelete),
+
+			Importer: &schema.ResourceImporter{
+				StateContext: schema.ImportStatePassthroughContext,
+			},
+
+			Schema: map[string]*schema.Schema{
+				"check_id": {
+					Description: "The ID of the check to manage alerts for.",
+					Type:        schema.TypeInt,
+					Required:    true,
+					ForceNew:    true,
+				},
+				"alerts": {
+					Description: "List of alerts for the check.",
+					Type:        schema.TypeSet,
+					Required:    true,
+					ConfigMode:  schema.SchemaConfigModeAttr,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"name": {
+								Description: "Name of the alert. Must be one of: " + AlertNameProbeFailedExecutionsTooHigh + ", " + AlertNameTLSTargetCertificateCloseToExpiring,
+								Type:        schema.TypeString,
+								Required:    true,
+								ValidateFunc: validation.StringInSlice([]string{
+									AlertNameProbeFailedExecutionsTooHigh,
+									AlertNameTLSTargetCertificateCloseToExpiring,
+								}, false),
+							},
+							"threshold": {
+								Description: "Threshold value for the alert.",
+								Type:        schema.TypeFloat,
+								Required:    true,
+							},
+							"period": {
+								Description: "Period for the alert threshold. Required only for " + AlertNameProbeFailedExecutionsTooHigh + " alerts. One of: `1m`, `2m`, `5m`, `10m`, `15m`, `20m`, `30m`, `1h`.",
+								Type:        schema.TypeString,
+								Required:    false,
+								Optional:    true,
+								Default:     "",
+								ValidateFunc: validation.StringInSlice([]string{
+									"", "1m", "2m", "5m", "10m", "15m", "20m", "30m", "1h",
+								}, false),
+							},
+						},
+					},
+				},
+			},
+		},
+	)
+}
+
+func resourceCheckAlertCreate(ctx context.Context, d *schema.ResourceData, c *smapi.Client) diag.Diagnostics {
+	checkID := int64(d.Get("check_id").(int))
+
+	// Validate that the check exists
+	if err := validateCheckExists(ctx, c, checkID); err != nil {
+		return diag.FromErr(err)
+	}
+
+	alerts, err := makeCheckAlerts(d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	_, err = c.UpdateCheckAlerts(ctx, checkID, alerts)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(fmt.Sprintf("%d", checkID))
```

Comment:
```suggestion
	d.SetId(strconv.FormatInt(checkID, 10))
```

Beyond that suggestion, I'm a little bit confused on this, probably because I don't know much about the Terraform provider implementation or how it actually works. In this function we receive a `ResourceData` that I understand matches the definition written by the user in the `.tf` file. So we expect that definition to contain a `check_id` as defined in the docs. Is this correct?

But then here we set the ID to the same check_id. I guess we have to do that due to how Terraform works? Reading the doc for the `SetId` method:
```
// SetId sets the ID of the resource. If the value is blank, then the
// resource is destroyed.
```

Reply:
> Beyond that suggestion, I'm a little bit confused on this, probably because I don't know much about the Terraform provider implementation or how it actually works. In this function we receive a ResourceData that I understand matches the definition written by the user in the .tf file. So we expect that definition to contain a check_id as defined in the docs. Is this correct?
But then here we set the ID to the same check_id. I guess we have to do that due to how Terraform works? Reading the doc for the SetId method

Yes, that's correct. The `ResourceData` object represents the resource configuration as defined in the `.tf` file. When we create a resource, we need to set it an ID which Terraform uses to track the resource in the state file. I'm setting the check_id as its ID as it's unique, and those IDs are scoped to their resource type (so different resource types can use the same ID value without conflict).

As an example, this is the state entry from `terraform.tfstate` that was generated for the the alerts resource I created when testing these changes:
![image](https://github.com/user-attachments/assets/0ffdb3b1-4e46-43e7-bff5-3bf5806d2a0b)

And this is the resource definition
![image](https://github.com/user-attachments/assets/eb40160e-76ff-4f85-83ab-e900c4dd5594)

You can notice how the generated state entry contains a `check_id` plus an `id` property.